### ISSUE TITLE
Question regarding __riscv_vsetvl mask/tail policy

### ISSUE THREAD

**Issue by OMaghiarIMG** [riscv-non-isa/rvv-intrinsic-doc#247]
Hello,

I was looking at an example in which we use _tu intrinsics:
![image](https://github.com/riscv-non-isa/rvv-intrinsic-doc/assets/118359779/af276ee8-6025-4796-80ed-9cd9cc6abafe)

The first __riscv_vsetvl translates to vsetvli ta,mu, then __riscv_vle32 adds another vsetvl to set the policy to tu,ma.
Would it be possible to have a single vsetvl with tu,ma since the beginning?

Would having _mu/_tu variants of the __riscv_vsetvl intrinsic make sense?

**Comment by zhongjuzhe** [comment]
No, I don't think we need to add _mu/_tu variants of __riscv_vsetvl intrinsic.

__riscv_vsetvl intrinsic is defined as any tail policy and any mask policy.

Compiler should be able to well optimize vsetvlis.

Compiler can well handle your case, no redundant vsetvli:

https://godbolt.org/z/jMra9dMKG

**Comment by OMaghiarIMG** [comment]
Thanks, yes this issue was present on my local clang16, didn't look at gcc.
Here's an example with a mask setting instruction, gcc sets tu,ma since the beginning, clang does a switch to ta and then back to tu.
https://compiler-explorer.com/z/TE5fzEh7j

**Comment by zhongjuzhe** [comment]
I think you should try trunk Clang.

The intrinsic has significantly changed recently and will be frozen soon.
Clang16 didn't support the latest rvv intrinsics.
